Client/Frontend/Browser/TabManager.swift

```diff
@@ -453,14 +453,39 @@ class TabManager: NSObject {
                     for delegate in self.delegates {
                         delegate.get()?.tabManagerDidAddTabs(self)
                     }
+                } else {
+                    self.showFocusPromoToast()
                 }
                 self.eraseUndoCache()
             })
         }
 
         delegates.forEach { $0.get()?.tabManagerDidRemoveAllTabs(self, toast: toast) }
     }
-    
+
+    func showFocusPromoToast() {
+        // add check for event stuff
+        guard let shouldShowFocusUI = LeanplumIntegration.sharedInstance.getBoolVariableFromServer(key: "shouldShowFocusUI"), let focus = NSURL(string: "firefox-focus://"), let klar = NSURL(string: "firefox-klar://"), !UIApplication.shared.canOpenURL(focus as URL) && !UIApplication.shared.canOpenURL(klar as URL) && shouldShowFocusUI else {
+            return
+        }
+
+        var toast: LeanplumPromoButtonToast
+        let templateDict = LeanplumIntegration.sharedInstance.getTemplateDictionary()!
```

Comment:
Please change this into an `if let` so that we can get rid of the explicit unwrap.